### PR TITLE
Create Board routes

### PR DESCRIPTION
**Pull request by Rixikun**
### Added
- GET /:id
- POST /
- PATCH /:id
- DELETE /:id
- url endpoint: "/api/board"
- BoardSchema.statics.findBoard()

( some lines cleaned by Prettier )

### CODE REVIEW

File: server/models/Board.js [hunk 1]
```diff
@@ -43,6 +43,11 @@ BoardSchema.statics.deleteBoard = function (boardId) {
   this.deleteOne({ _id: mongoose.Types.ObjectId(boardId) }).exec();
 };
 
+// gets a Board by _id
+BoardSchema.statics.findBoard = function (boardId) {
+  return this.find({ _id: mongoose.Types.ObjectId(boardId) });
```
Comment (dproc96) [inline server/models/Board.js]: You can just use findById for this, FYI

Comment (dproc96) [reply server/models/Board.js]: You'll also probably want to populate the columns and cards in this function

**Review by dproc96 — COMMENTED** [review]
Small update request to populate the data in the board, you can test in Postman if this is working when you make the change

File: server/models/Board.js (hunk 1 above)
Comment (Rixikun) [reply server/models/Board.js]: okay!

File: server/models/Board.js [hunk 2]
```diff
@@ -43,6 +43,37 @@ BoardSchema.statics.deleteBoard = function (boardId) {
   this.deleteOne({ _id: mongoose.Types.ObjectId(boardId) }).exec();
 };
 
+// gets a Board by _id
+BoardSchema.statics.findBoard = async function (boardId) {
+  const foundBoard = await this.findById(boardId);
```
Comment (dproc96) [inline server/models/Board.js]:
```suggestion
  const foundBoard = await this.findById(boardId).populate({ 
     path: 'columns',
     populate: {
       path: 'cards',
       model: 'Card'
     } 
  });
```
This is a much easier way to accomplish this, but clever implementation!

**Review by dproc96 — APPROVED** [review]
This looks good to go, we may need to update the PATCH route when integrating add column and add card but let's cross that bridge when we come to it

**Pull request merged**